packages/typespec-client-generator-core/test/decorators.test.ts

```diff
@@ -3785,4 +3785,334 @@ describe("typespec-client-generator-core: decorators", () => {
+      ok(containerNameParam);
+      strictEqual(containerNameParam.onClient, true);
+
+      const downloadOp = download.operation;
+      strictEqual(downloadOp.parameters.length, 2);
+      const blobNameOpParam = downloadOp.parameters[0];
+      strictEqual(blobNameOpParam.name, "blobName");
+      strictEqual(blobNameOpParam.correspondingMethodParams.length, 1);
+      strictEqual(blobNameOpParam.correspondingMethodParams[0], blobName);
+
+      const containerNameOpParam = downloadOp.parameters[1];
+      strictEqual(containerNameOpParam.name, "containerName");
+      strictEqual(containerNameOpParam.correspondingMethodParams.length, 1);
+      strictEqual(containerNameOpParam.correspondingMethodParams[0], containerName);
+    });
+
+    it("@operationGroup without same model on parent client", async () => {
+      const diagnostics = await runner.diagnose(
+        `
+        @service
+        namespace MyService;
+
+        @operationGroup
+        interface MyInterface {
+          op download(@path blobName: string, @path containerName: string): void;
+        }
+
+        model MyClientInitialization {
+          blobName: string;
+          containerName: string;
+        }
+
+        @@clientInitialization(MyService.MyInterface, MyClientInitialization);
+        `
+      );
+      expectDiagnostics(diagnostics, {
+        code: "@azure-tools/typespec-client-generator-core/assigning-public-params-to-internal-client",
+      });
+    });
+
+    it("@paramAlias", async () => {
+      await runner.compileWithCustomization(
+        `
+        @service
+        namespace MyService;
+
+        op download(@path blob: string): void;
+        op upload(@path blobName: string): void;
+        `,
+        `
+        namespace MyCustomizations;
+
+        model MyClientInitialization {
+          @paramAlias("blob")
+          blobName: string;
+        }
+
+        @@clientInitialization(MyService, MyCustomizations.MyClientInitialization);
+        `
+      );
+      const sdkPackage = runner.context.sdkPackage;
+      const client = sdkPackage.clients[0];
+      strictEqual(client.initialization.properties.length, 2);
+      const endpoint = client.initialization.properties.find((x) => x.kind === "endpoint");
+      ok(endpoint);
+      const blobName = client.initialization.properties.find((x) => x.name === "blobName");
+      ok(blobName);
+      strictEqual(blobName.clientDefaultValue, undefined);
+      strictEqual(blobName.onClient, true);
+      strictEqual(blobName.optional, false);
+
+      const methods = client.methods;
+      strictEqual(methods.length, 2);
+      const download = methods[0];
+      strictEqual(download.name, "download");
+      strictEqual(download.kind, "basic");
+      strictEqual(download.parameters.length, 1);
+
+      const downloadBlobNameParam = download.parameters.find((x) => x.name === "blobName");
+      ok(downloadBlobNameParam);
+      strictEqual(downloadBlobNameParam.onClient, true);
```

Comment:
From this test, it appears that TCGC automatically rename the `blob` in `download` to `blobName`, according to `@paramAlias`?

So emitter would only need to map it by name.

Is it correct understanding?

Reply:
We may need to be extra careful, if service write e.g.

```
model RequestBody {
  blobName: string;
  otherProperties: ...;
}

op action(@path blob: string, ...RequestBody);
```

Reply:
This is because we have a bug where we're still adding some client params to the method signature. That fix has been merged and I'm going to fix the test in this PR. Good catch!